### PR TITLE
llvm-core: unpin cmake version

### PR DESCRIPTION
**Pull request by pierricgimmig**
In llvm-core, the cmake version was pinned to 3.20.5, making it impossible to build with VS 2022 on Windows since VS 2022 generators were introduced with cmake 3.21. From the original comment, it seems the intention was not to pin cmake, but rather specify a minimum version.

This change simply transforms the cmake pinning to a mere minimum requirement, making it possible to compile llvm-core with toolchains born after cmake 3.20.5.

Specify library name and version:  **llvm-core/all**

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [ ] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/llvm-core/all/conanfile.py
Comment (RubenRBS) [inline recipes/llvm-core/all/conanfile.py:80]:
```suggestion
        'cmake/[>=3.20.5 <4]'
```

Just in case :)

**Review by RubenRBS — DISMISSED** [review]
Hi! Thanks a lot for taking the time to fix this issue, we really appreciate it

Please remember to request access by writing in https://github.com/conan-io/conan-center-index/issues/4 before we can merge this PR, after that the PR will be picked automatically once we leave maintenance mode in the CI systems, so no need for further actions on your part :)

**Comment by pierricgimmig** [comment]
Hi @RubenRBS , do I need to do somehting more on my end? Thanks!

**Comment by jcar87** [comment]
Thanks @pierricgimmig  and apologies for the delay on this.
I believe this no longer applies, as the graphviz output is no longer used so there is no strict requirement on any cmake version now: https://github.com/conan-io/conan-center-index/blob/master/recipes/llvm-core/all/conanfile.py

Closing this as no longer needed - but please let us know if there are any issues.

**Pull request closed without merging**
